OSDOCS#4603: Adds the overview page for Network Observability (#53360)
openshift/openshift-docs: networking/network_observability/network-observability-overview.adoc

:_content-type: ASSEMBLY
[id='network-observability-overview']
= About Network Observability
include::_attributes/common-attributes.adoc[]
:context: network-observability-overview

toc::[]

> Red Hat offers cluster administrators the Network Observability Operator to observe the network traffic for {product-title} clusters.

Reviewer: @jotak -  I suggest adding some of the most common use cases that customers are expected to benefit from when using the functionality of this operator.

Reviewer: Trying in slack

[id="dependencies-network-observability"]
== Dependencies of Network Observability Operator

> The Network Observability Operator requires the following Operators:

Reviewer: @stleerh --> I suggest splitting this section between dependencies or maybe even better to call those "prerequisites" and between other "Optional components" (the later are much less critical and can move down in the text)

Reviewer 2: +1: only mention Loki as a requirement
And below, maybe a subsection named "Optional dependencies" that list Kafka and Grafana

Author: Addressed

> The Network Observability Operator (NOO) requires the following Operators:

Reviewer 2: Not sure if we want to mention the NOO acronym here? It's mostly an internal name. Unless it helps for the doc itself?

Author: Right. I removed the acronym.

> * Loki Operator: You must install Loki install before you install the Network Observability Operator. The Loki Operator is the backend that you can use to store all collected flows.

Reviewer 3: ```suggestion
* Loki Operator: You must provide a dedicated Loki instance before you install the Network Observability Operator. Loki is the backend that you can use to store all collected flows.
```

Reviewer: @jpinsonneau, if we do that, my suggestion is far down in the text in the "small letters" section and not at the top.

Reviewer 2: Well, we should not say that there's a dependency on "Loki Operator", but just on "Loki", and then we can mention the operator as a recommended way to install it

Author: Addressed- mentioned that it is recommended to install Loki Operator

> * Loki Operator: You must install Loki install before you install the Network Observability Operator. The Loki Operator is the backend that you can use to store all collected flows.

Author: Addressed

> * Grafana Operator: You can optionally install Grafana Operator for using custom dashboards and querying capabilities.

Author: Addressed- mentioned that it is unsupported

> * Kafka Operator: You can optionally install Kafka Operator. The Kafka Operator provides scalability, resiliency and high availability in the {product-title} cluster.

Reviewer 3: ```suggestion
* Red Hat Integration - AMQ Streams: You can optionally install Kafka using AMQ Streams. This Operator provides scalability, resiliency and high availability in the {product-title} cluster.
```

Reviewer 3: `Red Hat Integration - AMQ Streams` is downstream name
`Strimzi` is the upstream one

Both are Kafka operators for OCP

Author: Addressed- added installing kafka operator using AMQ streams as a recommendation

[id="network-observability-operator"]
== Network Observability Operator

> The Network Observability Operator deploys a monitoring pipeline to collect and enrich network flows. The NetObserv eBPF agent or CNI can create network flows. You can export these network flows in IPFIX format, such as OVN-Kubernetes.

Reviewer 3: ```suggestion
The Network Observability Operator deploys a monitoring pipeline to collect and enrich network flows.
```

I would avoid these eBPF / IPFIX mention at this stage:
- we only support eBPF agent downstream
- you can use IPFIX export with OVN Kubernetes only and without RH support

What do you think ?

Author: Addressed

> The Network Observability Operator provides dashboards, metrics, and keeps flows accessible in a log store, Grafana, and Loki. You can access these dashboards in the {product-title} console.

Reviewer 3: ```suggestion
The Network Observability Operator provides dashboards, metrics, and keeps flows accessible in Grafana Loki log store. You can access these dashboards in the {product-title} console.
```

Be carefull [Grafana](https://grafana.com/grafana/) != [Loki](https://grafana.com/oss/loki/) but both are Grafana products :exploding_head:

Reviewer 4: Suggest:  
Network Observability provides three views of your network traffic: metrics, topology, and table information.  
  
I've already mentioned about storage in my previous comment.

Reviewer 2: @stleerh there's already these three other PRs about it: https://github.com/openshift/openshift-docs/pull/53405 , https://github.com/openshift/openshift-docs/pull/53366 and https://github.com/openshift/openshift-docs/pull/53362
Maybe just put links to these sections

[id="network-observability-dashboards"]
== Network Observability metrics

By using Network Observability metrics you can refine the stats by using the filters and display options. You can access the information for node, namespace, owner, pod, and service.

[id="no-console-integration"]
== {product-title} console integration

{product-title} console integration offers topology view and Netflow tables.

[id="network-observability-topology-views"]
=== Network Observability topology views

> The topology view represents traffic between the {product-title} components as a graph. You can refine the stats by using the filters and display options. You can access the information for node, namespace, owner, pod, and service.

Reviewer 3: ```suggestion
The topology view represents traffic between the {product-title} components as a network graph. You can refine the graph by using the filters and display options. You can access the information for node, namespace, owner, pod, and service.
```

Author: Addressed

> The topology view represents traffic between the {product-title} components as a graph. You can refine the stats by using the filters and display options. You can access the information for node, namespace, owner, pod, and service.

Reviewer 4: Have three sections, one on metrics, another on topology, and a third for table.  
  
In the latest UI (see https://github.com/netobserv/network-observability-operator/blob/main/docs/assets/overview-dashboard.png), there are three tabs (not two).

Reviewer 2: @stleerh there's already these three other PRs about it: https://github.com/openshift/openshift-docs/pull/53405 , https://github.com/openshift/openshift-docs/pull/53366 and https://github.com/openshift/openshift-docs/pull/53362
Maybe just put links to these sections

[id="netflow-tables"]
=== NetFlow tables

The NetFlow table view provides a view for raw flows, non aggregated filtering options, and configurable columns.

> 

Reviewer: I miss a section with links and references to additional materials i.e. quick starting guide, github organization, sizing guide etc.

Author: I will be adding the direct links to the actual sections once other PRs are merged

:_content-type: ASSEMBLY

> [id='network-observability-overview']

Reviewer 5: Use double quotes `"` for consistency. Not sure whether it has any other consequence though.

Author: Acked

= About Network Observability
include::_attributes/common-attributes.adoc[]
:context: network-observability-overview

toc::[]

> Red Hat offers cluster administrators the Network Observability Operator to observe the network traffic for {product-title} clusters. The NetObserv eBPF agent uses eBPF technology to create network flows based on the IPFIX format. Alternatively, you can export network flows if you are using OVN Kubernetes, but this is not supported.  The network flow data is saved to an object store. You can access the network flow through Loki by using Web console or Grafana

Reviewer 6: I think this section is confusing and needs to be rewritten.  We should really call out the value here.   I think it's confusing to talk about eBPF and IPFIX in the same sentence.
I also think the sentence starting "Alternatively" doesn't add much value -- why would we say you can do something but it's not supported -- in the first thing customers see.  
I also think we should probably not mention Grafana.  In this section we should say that Netflow information is available in the OpenShift Console.

Reviewer 7: Per discussion with @mcurry-rh it is essential for us to highlight eBPF and how Network Observability Operator is using it. Thanks!

Author: Addressed

> Red Hat offers cluster administrators the Network Observability Operator to observe the network traffic for {product-title} clusters. The NetObserv eBPF agent uses eBPF technology to create network flows based on the IPFIX format. Alternatively, you can export network flows if you are using OVN Kubernetes, but this is not supported.  The network flow data is saved to an object store. You can access the network flow through Loki by using Web console or Grafana

Reviewer 6: I would just say something like this.  Network Observability uses eBPF technology to create network flows based on the IPFIX format.  The network flows are then enriched and with kubernetes (or would it be openshift???) information and then stored.  Stored netflow information can be viewed and analyzed in the OpenShift console for further insight, troubleshooting and analysis.

> Red Hat offers cluster administrators the Network Observability Operator to observe the network traffic for {product-title} clusters. The NetObserv eBPF agent uses eBPF technology to create network flows based on the IPFIX format. Alternatively, you can export network flows if you are using OVN Kubernetes, but this is not supported.  The network flow data is saved to an object store. You can access the network flow through Loki by using Web console or Grafana

Reviewer 8: I've received feedback that we want to de-emphasize IPFIX, as it is unsupported by Red Hat as @jpinsonneau  pointed out to me in my installation docs PR ([review comment](https://github.com/openshift/openshift-docs/pull/53263#discussion_r1038226928)).

> Red Hat offers cluster administrators the Network Observability Operator to observe the network traffic for {product-title} clusters. The NetObserv eBPF agent uses eBPF technology to create network flows based on the IPFIX format. Alternatively, you can export network flows if you are using OVN Kubernetes, but this is not supported.  The network flow data is saved to an object store. You can access the network flow through Loki by using Web console or Grafana

Reviewer 6: yeah -- but i think it's ok to mention it's in ipfix format.

> Red Hat offers cluster administrators the Network Observability Operator to observe the network traffic for {product-title} clusters. The NetObserv eBPF agent uses eBPF technology to create network flows based on the IPFIX format. Alternatively, you can export network flows if you are using OVN Kubernetes, but this is not supported.  The network flow data is saved to an object store. You can access the network flow through Loki by using Web console or Grafana

Author: Addressed

[id="dependency-network-observability"]
== Dependency of Network Observability Operator

The Network Observability Operator requires the following Operators:

> * Loki: You must install Loki. Loki is the backend that is used to store all collected flows. It is recommended to install the Loki Operator for the installation of Network Observability Operator.

Reviewer 5: 2 conflicting statements here. I think you can remove `It is recommended to install the Loki Operator for the installation of Network Observability Operator.` Instead write `You must install the Loki Operator for the installation of Network Observability Operator.` in the beginning of the sentence.

Author: @snarayan-redhat 
Actually Loki the backend is mandatory but Loki Operator is not a must-have. That's why Developers suggested writing "It is recommended"

Reviewer 9: @xenolinux I tend to agree here. It is not clear that we recommend installing the Loki Operator to install Loki. Now, it seems like two steps: Install Loki; Install Loki Operator. Would this make sense?  
_It is recommended to install Loki by installing the Loki Operator._

Author: Addressed

> * Loki Operator: It is recommended to install the Loki Operator for the installation of Network Observability Operator. The Loki Operator is the backend that you can use to store all collected flows.

Reviewer 3: Yes, I would like personnaly, else you need to hack or skip gateway

> * Loki Operator: It is recommended to install the Loki Operator for the installation of Network Observability Operator. The Loki Operator is the backend that you can use to store all collected flows.

Reviewer 8: @memodi @jpinsonneau I also mention Loki version in the Operator installation prerequisites, ICYMI [here](https://53263--docspreview.netlify.app/openshift-enterprise/latest/networking/network_observability/installing-operators.html)

> * Loki Operator: It is recommended to install the Loki Operator for the installation of Network Observability Operator. The Loki Operator is the backend that you can use to store all collected flows.

Reviewer 2: I guess it's fine as it's already mentioned in the installation section

> * Loki Operator: It is recommended to install the Loki Operator for the installation of Network Observability Operator. The Loki Operator is the backend that you can use to store all collected flows.

Author: Acked

[id="optional-dependency-network-observability"]

> == Optional dependency of the Network Observability Operator

Reviewer 10: ```suggestion
== Optional dependencies of the Network Observability Operator
```

Author: Addressed

> * Grafana: You can install Grafana for using custom dashboards and querying capabilities, for instance using the Grafana Operator. Red Hat does not support Grafana Operator.

Reviewer 5: `for instance` can be removed

Author: Acked

> * Kafka Operator: It is recommended to install Kafka using AMQ Streams. This Operator provides scalability, resiliency and high availability in the {product-title} cluster.

Reviewer 6: I would lead with why kafka, and then the recommendation to use AMQ streams.
So just switch the sentence order.
It might be good to mention that it's intended for larger scale deployments/etc.

Author: Addressed

[id="network-observability-operator"]
== Network Observability Operator

> The Network Observability Operator deploys a monitoring pipeline to collect and enrich network flows. The NetObserv eBPF agent or CNI can create network flows.

Reviewer 6: I don't know what this means: "The NetObserv eBPF agent or CNI can create network flows."
Maybe change to "Network flows are created in Network Observability by an eBPF agent, deployed to each node via a daemonset"

Author: Addressed

> The Network Observability Operator deploys a monitoring pipeline to collect and enrich network flows. The eBPF agent is deployed as a `daemonset` object and creates network flows. The network flows are then collected and enriched with the Kubernetes metadata, before stored in Loki.

Reviewer 5: ```suggestion
The Network Observability Operator deploys a monitoring pipeline to collect and enrich the network flows. The eBPF agent, which is deployed as a `daemonset` object, creates the network flows. The network flows are then collected and enriched with the Kubernetes metadata, before storing in Loki.
```

Author: Acked

[id="no-console-integration"]
== {product-title} console integration

> {product-title} console integration offers overview, topology view and Netflow tables.

Reviewer 5: ```suggestion
{product-title} console integration offers overview, topology view and traffic flow tables.
```
The term Netflow is removed from UI perspective hence let's not use that in our doc.

Author: Acked

[id="network-observability-dashboards"]
=== Network Observability metrics

> By using Network Observability metrics you can refine the stats by using the filters and display options. The {product-title} console offers the *Overview* tab which displays the overall, aggregated metrics of the network traffic flow on the cluster. You can access the information for node, namespace, owner, pod, and service.

Reviewer 10: ```suggestion
By using Network Observability metrics you can refine the stats by using the filters and display options. The {product-title} console offers the *Overview* tab which displays the overall aggregated metrics of the network traffic flow on the cluster. You can access the information for node, namespace, owner, pod, and service.
```

Author: Addressed

> By using Network Observability metrics you can refine the stats by using the filters and display options. The {product-title} console offers the *Overview* tab which displays the overall aggregated metrics of the network traffic flow on the cluster. You can access the information for node, namespace, owner, pod, and service.

Reviewer 5: ```suggestion
By using Network Observability metrics, you can refine the stats by using the filters and display options. The {product-title} console offers the *Overview* tab which displays the overall aggregated metrics of the network traffic flow on the cluster. You can access the information for node, namespace, owner, pod, and service.
```

Author: Acked

[id="network-observability-topology-views"]
=== Network Observability topology views

> The topology view represents traffic between the {product-title} components as a network graph. You can refine the graph by using the filters and display options. You can access the information for node, namespace, owner, pod, and service. The {product-title} console offers the *Topology* tab which displays a graphical representation of the network flows and the amount of traffic.

Reviewer 5: ```suggestion
The {product-title} console offers the *Topology* tab which displays a graphical representation of the network flows and the amount of traffic. The topology view represents traffic between the {product-title} components as a network graph. You can refine the graph by using the filters and display options. You can access the information for node, namespace, owner, pod, and service. 
```

Author: Acked

[id="netflow-tables"]

> === NetFlow tables

Reviewer 5: As in my previous comment, this term is not present in UI, it's something specific to dev team. So let's get rid of this term. It's called `Traffic flows` in UI. So you can name it accordingly

Author: Acked
